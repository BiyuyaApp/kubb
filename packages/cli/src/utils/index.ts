export * from './getConfig.ts'
export * from './getCosmiConfig.ts'
export * from './getErrorCause.ts'
export * from './getPlugins.ts'
export * from './OraWritable.ts'
export * from './parseHrtimeToSeconds.ts'
export * from './renderErrors.ts'
export * from './spinner.ts'
export * from './watcher.ts'
