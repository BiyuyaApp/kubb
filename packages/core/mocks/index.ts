import { pascalCase } from '../src/transformers/casing.ts'

import { PluginManager } from '../src/PluginManager.ts'
export const mockedPluginManager = {
  resolveName: ({ name, type }) => {
    if (type === 'type') {
      return pascalCase(name)
    }

    return name
  },
  resolvePath: ({ baseName }) => baseName,
  logger: {
    info(message) {
      console.log(message)
    },
    error(message) {
      console.log(message)
    },
    warn(message) {
      console.log(message)
    },
    log(message) {
      console.log(message)
    },
  },
} as PluginManager
