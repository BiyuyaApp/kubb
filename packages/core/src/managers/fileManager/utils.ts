import uniq from 'lodash.uniq'

import type { File } from './types'

export function combineFiles(files: Array<File | null>) {
  return files.filter(Boolean).reduce((acc, curr: File | null) => {
    if (!curr) {
      return acc
    }
    const prevIndex = acc.findIndex((item) => item.path === curr.path)

    if (prevIndex !== -1) {
      const prev = acc[prevIndex]
      acc[prevIndex] = {
        ...curr,
        source: `${prev.source}\n${curr.source}`,
        imports: [...(prev.imports || []), ...(curr.imports || [])],
      }
    } else {
      acc.push(curr)
    }

    return acc
  }, [] as File[])
}

export function getFileSource(file: File) {
  // TODO make generic check
  if (!file.fileName.endsWith('.ts')) {
    return file.source
  }
  const imports: File['imports'] = []

  file.imports?.forEach((curr) => {
    const exists = imports.find((imp) => imp.path === curr.path)
    if (!exists) {
      imports.push({
        ...curr,
        name: Array.isArray(curr.name) ? uniq(curr.name) : curr.name,
      })
    }

    if (exists && !Array.isArray(exists.name) && exists.name !== curr.name) {
      imports.push(curr)
    }

    if (exists && Array.isArray(exists.name)) {
      if (Array.isArray(curr.name)) {
        exists.name = uniq([...exists.name, ...curr.name])
      }
    }
  })

  const importSource = imports.reduce((prev, curr) => {
    if (Array.isArray(curr.name)) {
      return `${prev}\nimport ${curr.isTypeOnly ? 'type ' : ''}{ ${curr.name.join(', ')} } from "${curr.path}";`
    }

    return `${prev}\nimport ${curr.isTypeOnly ? 'type ' : ''}${curr.name} from "${curr.path}";`
  }, '')

  if (importSource) {
    return `${importSource}\n${file.source}`
  }

  return file.source
}
