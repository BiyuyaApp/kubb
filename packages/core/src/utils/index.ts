export * from './isPromise'
export * from './write'
export * from './cache'
export * from './read'
export * from './isURL'
export * from './objectToParameters'
export * from './nameSorter'
export * from './jsdoc'
export * from './getUniqueName'
