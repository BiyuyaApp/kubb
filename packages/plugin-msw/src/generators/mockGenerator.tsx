import { pluginFakerName } from '@kubb/plugin-faker'
import { createReactGenerator } from '@kubb/plugin-oas'
import { useOperationManager } from '@kubb/plugin-oas/hooks'
import { pluginTsName } from '@kubb/plugin-ts'
import { File, useApp } from '@kubb/react'
import { Mock } from '../components'
import type { PluginMsw } from '../types'

export const mockGenerator = createReactGenerator<PluginMsw>({
  name: 'plugin-msw',
  Operation({ operation }) {
    const { pluginManager, plugin } = useApp<PluginMsw>()
    const { getSchemas, getName, getFile } = useOperationManager()

    const name = getName(operation, { type: 'function' })
    const file = getFile(operation)
    const fakerFile = getFile(operation, { pluginKey: [pluginFakerName] })
    const fakerSchemas = getSchemas(operation, { pluginKey: [pluginTsName] })

    const responseName = pluginManager.resolveName({
      pluginKey: [pluginFakerName],
      name: fakerSchemas.response.name,
      type: 'function',
    })

    return (
      <File baseName={file.baseName} path={file.path} meta={file.meta}>
        <File.Import name={['http']} path={'msw'} />
        {fakerFile && responseName && <File.Import extName={plugin.output?.extName} name={[responseName]} root={file.path} path={fakerFile.path} />}

        <Mock name={name} responseName={responseName} operation={operation} />
      </File>
    )
  },
})
