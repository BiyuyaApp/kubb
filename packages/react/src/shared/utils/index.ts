export * from './squashExportNodes.ts'
export * from './squashImportNodes.ts'
export * from './squashTextNodes.ts'
export * from './getFile.ts'
