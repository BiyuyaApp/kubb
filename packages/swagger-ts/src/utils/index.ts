export * from './keywordTypeNodes.ts'
export * from './resolve.ts'
