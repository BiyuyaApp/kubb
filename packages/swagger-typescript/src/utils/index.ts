export * from './print'
export * from './jsdoc'
export * from './write'
