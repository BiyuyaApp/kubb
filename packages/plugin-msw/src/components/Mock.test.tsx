import { matchFiles, mockedPluginManager } from '@kubb/core/mocks'

import type { Plugin } from '@kubb/core'
import { OperationGenerator } from '@kubb/plugin-oas'
import { parseFromConfig } from '@kubb/plugin-oas/utils'

import { mockGenerator } from '../generators/mockGenerator.tsx'
import type { PluginMsw } from '../types.ts'

describe('<Mock/>', async () => {
  const oas = await parseFromConfig({
    root: './',
    output: { path: 'test', clean: true },
    input: { path: 'packages/plugin-msw/mocks/petStore.yaml' },
  })

  const options: PluginMsw['resolvedOptions'] = {
    dataReturnType: 'data',
    pathParamsType: 'object',
    baseURL: '',
  }
  const plugin = { options } as Plugin<PluginMsw>
  const og = new OperationGenerator<PluginMsw>(options as any, {
    oas,
    exclude: [],
    include: undefined,
    pluginManager: mockedPluginManager,
    plugin,
    contentType: undefined,
    override: undefined,
    mode: 'split',
  })

  test('showPetById', async () => {
    const operation = oas.operation('/pets/{petId}', 'get')
    const files = await mockGenerator.operation?.({
      operation,
      options,
      instance: og,
    })

    await matchFiles(files)
  })

  test('pets', async () => {
    const operation = oas.operation('/pets', 'post')
    const files = await mockGenerator.operation?.({
      operation,
      options,
      instance: og,
    })

    await matchFiles(files)
  })
})
