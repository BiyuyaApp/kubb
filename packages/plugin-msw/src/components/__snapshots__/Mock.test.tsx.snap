// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Mock/> > pets 1`] = `
"export const createPets = http.post('*/pets', function handler(info) {
    return new Response(JSON.stringify(createPetsMutationResponse()), {
      headers: {
        'Content-Type': 'application/json',
      },
    })
  })"
`;

exports[`<Mock/> > showPetById 1`] = `
"export const showPetById = http.get('*/pets/:petId', function handler(info) {
    return new Response(JSON.stringify(showPetByIdQueryResponse()), {
      headers: {
        'Content-Type': 'application/json',
      },
    })
  })"
`;
